src/main/java/org/sciborgs1155/robot/vision/Vision.java

```diff
@@ -0,0 +1,157 @@
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+import org.photonvision.PhotonPoseEstimator.PoseStrategy;
+import org.photonvision.simulation.PhotonCameraSim;
+import org.photonvision.simulation.SimCameraProperties;
+import org.photonvision.simulation.VisionSystemSim;
+import org.sciborgs1155.robot.Robot;
+
+public class Vision {
+  private final PhotonCamera frontCamera;
+  private final PhotonCamera rearCamera;
+  private final PhotonPoseEstimator frontPhotonEstimator;
+  private final PhotonPoseEstimator rearPhotonEstimator;
+  private double lastEstTimestamp = 0;
+
+  // Simulation
+  private PhotonCameraSim cameraSim;
+  private VisionSystemSim visionSim;
+
+  public Vision() {
+    this.frontCamera = new PhotonCamera(VisionConstants.FRONT_CAMERA_NAME);
+    this.rearCamera = new PhotonCamera(VisionConstants.REAR_CAMERA_NAME);
+
+    frontPhotonEstimator =
+        new PhotonPoseEstimator(
+            VisionConstants.TAG_LAYOUT,
+            PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+            frontCamera,
+            VisionConstants.ROBOT_TO_CAM);
+
+    frontPhotonEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
+    rearPhotonEstimator =
+        new PhotonPoseEstimator(
+            VisionConstants.TAG_LAYOUT,
+            PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+            rearCamera,
+            VisionConstants.ROBOT_TO_CAM);
+    rearPhotonEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
+
+    // ----- Simulation
+    if (Robot.isSimulation()) {
+      // Create the vision system simulation which handles cameras and targets on the field.
+      visionSim = new VisionSystemSim("main");
+      // Add all the AprilTags inside the tag layout as visible targets to this simulated field.
+      visionSim.addAprilTags(VisionConstants.TAG_LAYOUT);
+      // Create simulated camera properties. These can be set to mimic your actual camera.
+      var cameraProp = new SimCameraProperties();
+      cameraProp.setCalibration(960, 720, Rotation2d.fromDegrees(90));
+      cameraProp.setCalibError(0.35, 0.10);
+      cameraProp.setFPS(15);
+      cameraProp.setAvgLatencyMs(50);
+      cameraProp.setLatencyStdDevMs(15);
+      // Create a PhotonCameraSim which will update the linked PhotonCamera's values with visible
+      // targets.
+      cameraSim = new PhotonCameraSim(frontCamera, cameraProp);
+      // Add the simulated camera to view the targets on this simulated field.
+      visionSim.addCamera(cameraSim, VisionConstants.ROBOT_TO_CAM);
+
+      cameraSim.enableDrawWireframe(true);
+    }
+  }
+
+  /**
+   * The latest estimated robot pose on the field from vision data. This may be empty. This should
+   * only be called once per loop.
+   *
+   * @return An {@link EstimatedRobotPose} with an estimated pose, estimate timestamp, and targets
+   *     used for estimation.
+   */
+  public Optional<EstimatedRobotPose> getEstimatedGlobalPose(
```

Comment:
you can get rid of this method, since `getEstimatedGlobalPoses` is the same thing, but works for multiple cameras